From 0aed21bd6cdc3bc36f0f34e49783c83ae40357e1 Mon Sep 17 00:00:00 2001
From: Linh Nguyen <nvl1109@gmail.com>
Date: Sat, 23 Nov 2019 07:07:39 +0700
Subject: [PATCH] Make graphic libs configurable

---
 src/dispatch_common.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/src/dispatch_common.c b/src/dispatch_common.c
index 72cda4c..8750abc 100644
--- a/src/dispatch_common.c
+++ b/src/dispatch_common.c
@@ -173,6 +173,9 @@
 
 #include "dispatch_common.h"
 
+#define xstr(a) str(a)
+#define str(a) #a
+
 #ifdef __APPLE__
 #define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
 #elif defined(__ANDROID__)
@@ -191,10 +194,14 @@
 #define GLES1_LIB "libGLES_CM.dll"
 #define GLES2_LIB "libGLESv2.dll"
 #else
-#define EGL_LIB "libEGL.so.1"
-#define GLES1_LIB "libGLESv1_CM.so.1"
-#define GLES2_LIB "libGLESv2.so.2"
-#define OPENGL_LIB "libOpenGL.so.0"
+//#define EGL_LIB "libEGL.so.1"
+//#define GLES1_LIB "libGLESv1_CM.so.1"
+//#define GLES2_LIB "libGLESv2.so.2"
+//#define OPENGL_LIB "libOpenGL.so.0"
+#define GLX_LIB xstr(GLX_LIB_NAME)
+#define EGL_LIB xstr(EGL_LIB_NAME)
+#define GLES1_LIB xstr(GLES1_LIB_NAME)
+#define GLES2_LIB xstr(GLES2_LIB_NAME)
 #endif
 
 #ifdef __GNUC__
-- 
2.17.1

